datamanager/pumper/pumper.go

```diff
@@ -0,0 +1,103 @@
+package pumper
+
+import (
+	"context"
+	"errors"
+	"github.com/bitleak/lmstfy/config"
+	"github.com/bitleak/lmstfy/datamanager/storage"
+	"github.com/bitleak/lmstfy/helper"
+	"github.com/go-redis/redis/v8"
+	"github.com/go-redsync/redsync/v4/redis/goredis/v8"
+	"github.com/sirupsen/logrus"
+	"time"
+
+	"github.com/bitleak/lmstfy/datamanager/storage/model"
+	"github.com/bitleak/lmstfy/engine"
+	"github.com/go-redsync/redsync/v4"
+)
+
+const (
+	DefaultStoragePumpPeriod = 10 * 60 // number of seconds, equals 10 minutes
+	LoopPumpLockName         = "LP_redsync"
+	MaxJobBatchSize          = 1000
+)
+
+type DefaultPumper struct {
+	redisClient       *redis.Client
+	shutdown          chan struct{}
+	storagePumpPeriod int64
+}
+
+func NewPumper(cfg *config.Config, poolConfig *config.RedisConf) (Pumper, error) {
+	if cfg == nil {
+		return nil, errors.New("failed to create new pumper, nil config")
+	}
+	client := helper.NewRedisClient(&cfg.AdminRedis, nil)
+	if client.Ping(context.Background()).Err() != nil {
+		return nil, errors.New("failed to create redis client for pumper")
+	}
+	period := poolConfig.StoragePumpPeriod
+	if period <= 0 {
+		period = DefaultStoragePumpPeriod
+	}
+	return &DefaultPumper{
+		redisClient:       client,
+		shutdown:          make(chan struct{}),
+		storagePumpPeriod: int64(period),
+	}, nil
+}
+
+// LoopPump periodically checks job data ready time and pumps jobs to engine
+func (dp *DefaultPumper) LoopPump(st storage.Storage, eng engine.Engine) {
```

Comment:
Need test cases for pumper

Reply:
LoopPump test is covered in engine test, since all the dependencies are initialized there.